cfme/utils/template/base.py

```diff
@@ -277,6 +278,38 @@ def deploy_template(self):
             template.deploy(**deploy_args)
         return True
 
+    @log_wrap("checksum verification")
+    def checksum_verification(self):
+        # Download checksum from url
+        checksum = None
+        try:
+            response = request.urlopen('{}/SHA256SUM'.format(self.image_url))
+            checksums = response.read().decode('utf-8')
+            for line in checksums.split("\n"):
+                if self.image_name in line:
+                    checksum = line.strip().split()[0]
+        except URLError:
+            logger.warn('Failed download of checksum using urllib')
+        if not checksum:
+            logger.warn('Failed to get checksum of image from url')
```

Comment:
I think this should `return False` at this point as well, instead of leaving it to return None at the bottom of the function.  At this point the 'step' has failed.

Reply:
Okay, when it returns False. Does it interrupt the job? Because this is not such a big deal to interrupt template upload.

Reply:
IIRC both False or None will trigger the job 'failing'. I think it should fail if the checksum doesn't match, forcing us to investigate why (we can always re-run)

Reply:
Actually, ignore me. Because this is called by download_image, it won't stop the job. download_image returning False/None will stop the upload, because that step is wrapped.